pytorch_lightning/loops/fit_loop.py

```diff
@@ -129,6 +129,10 @@ def done(self) -> bool:
         Returns True if trainer.should_stop was set (e.g. by early stopping)
         or if the maximum number of steps or epochs is reached.
         """
+        if self.max_epochs < 0 and self.max_steps is None:
+            # If max_epochs is negative and max_steps is not defined, disable automatic stopping
+            return False
+
```

Comment:
i think we should flesh this out further. Across all of (max_steps, max_epochs, max_time):
- What are new valid configurations? 
- Wha are new invalid configurations?
- What priority do we give among steps/epochs/time? 
- Do we need to adjust defaults anywhere that depend on max_steps or max_epochs?

Reply:
**What are new valid configurations?**
The only new valid configuration is `max_epochs` declared as a negative integer + `max_steps = None`.

**What are new invalid configurations?**
- `max_epochs` as a negative integer and `max_steps` is positive. This should be invalid because specifying `max_steps` means automatic stopping won't be disabled. This will result in `max_epochs` being honored and no training occurring.
- `max_steps` or `max_time` as a negative value. This should be invalid because only `max_epochs` should be allowed to disable automatic stopping to reduce confusion and additional logic needed to maintain multiple flags.
- Setting any of `max_epochs`, `max_steps`, or `max_time` to be zero. This will result in no training occurring and doesn't make sense.

**Are there configurations which aren't exceptional but which we should raise some sort of warning for?**
No. We should not allow any of the above configurations.

**What priority do we give among steps/epochs/time?**
We should prioritize whichever occurs first with a user-defined value. This means if the user defines `max_steps`, we should only stop at this value even if the default `max_epochs` value occurs first. Even if automatic stopping is disabled by setting `max_epochs < 0` and `max_steps = None`, we should still honor `max_time`.

**Do we need to adjust defaults anywhere that depend on max_steps or max_epochs?**
All the defaults should remain. `max_time` and `max_steps` both default to None. `max_epochs` defaults to `1000` and is the only variable of the three not set to `None`.

Reply:
> What priority do we give among steps/epochs/time?
We should prioritize whichever occurs first with a user-defined value. This means if the user defines max_steps, we should only stop at this value even if the default max_epochs value occurs first. Even if automatic stopping is disabled by setting max_epochs < 0 and max_steps = None, we should still honor max_time.

That seems a bit strange to me. Why discard one (max_epochs) but not the other (max_time)?

I agree on the user-defined values though.

Reply:
The main intention of this PR was to disable lightning automatically stopping training after a certain number of epochs/steps were reached even if the user didn't want this behavior. You might still want a model to train for a certain amount of time, but not be limited by the number of epochs/steps. For instance, if you wanted a model to train at most for two weeks, but you weren't sure how many epochs/steps it would be able to reach, you could disable automatic stopping for epochs/steps, but still specify a time constraint.

Currently `max_time` is not really interacting much with `max_epochs` or `max_steps`. Even if `max_time` is specified, `max_epochs` will still default to `1000`. Whichever of `max_time` or `max_epochs`/`max_steps` comes first will decide when the model stops.

You could also still achieve no stopping at all if you set `max_epochs < 0`, `max_steps = None`, and `max_time = None`. 

This seems to achieve the best result of:
- You can disable stopping completely
- You can train until a certain time limit (regardless of `max_epochs`/`max_steps`)